packages/presentation/lib/src/pages/login_page/bloc/login_bloc.dart

```diff
@@ -40,27 +51,39 @@ class _LoginBloc extends BlocImpl<BaseArguments, LoginData>
   final LoginGoogleUseCase loginGoogleUseCase;
   final LoginFaceBookUseCase loginFaceBookUseCase;
   final LogButtonUseCase logButton;
-  final LoginValidator loginValidator;
+  final ValidateLoginFormUseCase formValidator;
+  final GlobalKey<FormState> formKey;
+  final ResultToLocalizedMapper localizationResultMapper;
 
   _LoginBloc(
     this.loginWithEmailAndPass,
     this.loginGoogleUseCase,
     this.loginFaceBookUseCase,
     this.logButton,
-    this.loginValidator,
+    this.formValidator,
+    this.formKey,
+    this.localizationResultMapper,
   ) : super(LoginData.init());
 
+  @override
+  void init() {
+    emit(data: tile);
+    super.init();
+  }
+
   @override
   Future<void> auth() async {
-    emit(data: tile, isLoading: false);
-    if (!loginValidator(tile.login, tile.password)) {
-      emit(data: tile.copyWith(errorMessage: 'Fill in your login or password'));
-      return;
-    }
-    emit(data: tile, isLoading: true);
     logButton(EventName.emailAndPasswordBtn);
     final UserEmailPass user = UserEmailPass(tile.login, tile.password);
-    _tryLogin(await loginWithEmailAndPass(user));
+    final validationResult = formValidator(user);
+    final localizedResult = localizationResultMapper(validationResult);
+    emit(
+      data: tile.fromValidationResult(result: localizedResult),
+      isLoading: true,
+    );
+    if (formKey.currentState?.validate() ?? false) {
+      _tryLogin(await loginWithEmailAndPass(user));
+    }
```

Comment:
I think it's better to handle it via exception throwing. something like this

```dart
try {
   await formValidator(user); //added future to have more suitable work with errors
   await _tryLogin();
} catch (ValidationException e) {
  //handle it
}
```

think over this idea

Reply:
interesting approach =)